src/dune_rules/scope.ml

```diff
@@ -85,13 +85,13 @@ module DB = struct
       else Lib.DB.Resolve_result.not_found
     | Found lib -> Memo.return (Lib.DB.Resolve_result.found lib)
     | Deprecated_library_name lib ->
-      Memo.return (Lib.DB.Resolve_result.redirect_in_the_same_db lib)
+      Memo.return (Lib.DB.Resolve_result.redirect parent lib)
```

Comment:
I believe this change is just saving us 1 redirect, because the new public name will exist in the public libs DB. We'd redirect anyway at resolution time (following 2 redirects, now just 1)